ME-neuter tests fix suggestion

wiktormowinski opened the pull request:
-Changed the method test cases : MES001.001, MNE002.001, MNE003.001, MNE004.001 
  are performed.
-Created a keyword to perform this operation

macpijan commented on dasharo-security/me-keyword.resource:
```diff
@@ -0,0 +1,9 @@
+
+
+*** Keywords ***
+Check If ME Is Disabled
```
This file should go into `lib/`, see how other keywords are distributed

macpijan replied:
Please add documentation section, explaining what this keyword does, what it returns, what it means

macpijan replied:
Either add to some existing library, or create a new one, and include it in the keywords.robot

macpijan commented on dasharo-security/me-keyword.resource:
```diff
@@ -0,0 +1,9 @@
+*** Keywords ***
+Check If ME Is Disabled
+    ${out}=    Run Process    /usr/sbin/setpci    -s    16.0    40.L
+    RETURN    Should Not Be Equal As Strings    ${out.stdout}[3]    0
```
Adding some comment here would be useful - e.g. we check for bit number X which should inidcate something

macpijan replied:
Is this return value useful ? Do we rely on it somehow? I see the kwd is called, but we do not assign return value anywhere.

wiktormowinski replied:
now it has a use, everything is explained in comment which i also added

macpijan commented on dasharo-security/me-keyword.resource:
```diff
@@ -0,0 +1,9 @@
+
+*** Keywords ***
+Check If ME Is Disabled
+    ${out}=    Run Process    /usr/sbin/setpci    -s    16.0    40.L
```
Run Process executes program on your computer (where the `robot`) process is started, you should use `Execute Command In Terminal` instead, to run on the DUT

wiktormowinski replied:
fixed

mkopec commented on lib/me-keyword.resource:
```diff
@@ -0,0 +1,22 @@
+    ...    -Enabled for 0
+
+    ${out}=    Execute Command In Terminal    setpci -s 16.0 40.L
+    ${char}=    Evaluate    ${out}[3]
```
We're converting a 32bit integer to a string, and then getting the 3rd character from the left, a rather roundabout way to read bits 16:19 of a register :). Let's add a comment explaining what's going on here, something like:

```
# Current Operation Mode: bits 16:19 of register HFSTS1 (0x40)
```

wiktormowinski replied:
will the comment be sufficient then? if so here's the update: https://github.com/Dasharo/open-source-firmware-validation/pull/375/commits/e8738cef4b6e89953cb45972e3df68a60965bd21

mkopec commented on lib/me-keyword.resource:
```diff
@@ -0,0 +1,22 @@
+    ${out}=    Execute Command In Terminal    setpci -s 16.0 40.L
+    ${char}=    Evaluate    ${out}[3]
+    IF    '${char}' == '3'
+        ${result}=    Evaluate    'Disabled(soft)'
```
```suggestion
        ${result}=    Evaluate    'Disabled (Soft)'
```

Let's make this consistent with how the option is called in the UEFI setup menu, and implemented in lib/options

wiktormowinski replied:
done: https://github.com/Dasharo/open-source-firmware-validation/pull/375/commits/e8738cef4b6e89953cb45972e3df68a60965bd21

mkopec commented on lib/me-keyword.resource:
```diff
@@ -0,0 +1,22 @@
+    IF    '${char}' == '3'
+        ${result}=    Evaluate    'Disabled(soft)'
+    ELSE IF    '${char}' == '2'
+        ${result}=    Evaluate    'Disabled(HAP)'
```
a/a
```suggestion
        ${result}=    Evaluate    'Disabled (HAP)'
```

wiktormowinski replied:
changes included in: https://github.com/Dasharo/open-source-firmware-validation/pull/375/commits/e8738cef4b6e89953cb45972e3df68a60965bd21

macpijan commented:
@wiktormowinski I rebased and squashed it.

On which devices it has been tested?

Protectli VP6670 returns:

```
setpci -s 16.0 40.L
setpci: Warning: No devices selected for operation group 1.
```

Because of that, your keyword returns:

![image](https://github.com/user-attachments/assets/e426bf08-b28f-42b6-84ed-022d5238cf38)

Which means this solution is not perfect.

I guess, we should first check if the PCI device exists. And only if it exists, we can try to probe. If it does not exist, we should assume it is disabled?

wiktormowinski commented:
> @wiktormowinski I rebased and squashed it.
> 
> On which devices it has been tested?
> 
> Protectli VP6670 returns:
> 
> ```
> setpci -s 16.0 40.L
> setpci: Warning: No devices selected for operation group 1.
> ```
> 
> Because of that, your keyword returns:
> 
> ![image](https://private-user-images.githubusercontent.com/19353650/351110779-e426bf08-b28f-42b6-84ed-022d5238cf38.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE3MTc5MzMsIm5iZiI6MTcyMTcxNzYzMywicGF0aCI6Ii8xOTM1MzY1MC8zNTExMTA3NzktZTQyNmJmMDgtYjI4Zi00MmI2LTg0ZWQtMDIyZDUyMzhjZjM4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzIzVDA2NTM1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNmNzk1OTM4ZWEyZDc0ZmJkMmM5M2Y5MTIwYzQwNWNmZjg5NjEwYmIwNjdjNTMwM2M3ZGYxYWJjNDM3YzdiYmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.IJWA4-a2q5_aa9Dr8fh5suHFh6ifS163IJ0sSScPiPg)
> 
> Which means this solution is not perfect.
> 
> I guess, we should first check if the PCI device exists. And only if it exists, we can try to probe. If it does not exist, we should assume it is disabled?

@macpijan only VP2420 like the scope stated.
 I will look into it in the near future.

wiktormowinski commented:
https://github.com/Dasharo/open-source-firmware-validation/pull/403
